src/libs/vehicle/mavlink/vehicle.ts

```diff
@@ -773,6 +774,68 @@ export abstract class MAVLinkVehicle<Modes> extends Vehicle.AbstractVehicle<Mode
     sendMavlinkMessage(heartbeatMessage)
   }
 
+  /**
+   * Send system time from GCS
+   */
+  sendSystemTime(): void {
+    const systemTimeMessage: Message.SystemTime = {
+      type: MAVLinkType.SYSTEM_TIME,
+      time_unix_usec: Math.floor(performance.now() * 1000),
+      time_boot_ms: 0,
+    }
+    sendMavlinkMessage(systemTimeMessage)
+  }
+
+  /**
+   * Measure network latency to vehicle using TIMESYNC message
+   * @returns {Promise<void>}
+   */
+  async measureLatency(): Promise<void> {
+    const time_sent_ns = Math.floor(performance.now()) * 1e6
```

Comment:
```suggestion
    const time_sent_ns = Math.floor(performance.now() * 1e6)
```
Rounding before scaling loses precision.